src/transformStateWithClones.js

```diff
@@ -6,8 +6,66 @@
  *
  * @return {Object[]}
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+
+const state = {
+  foo: 'bar',
+  name: 'Jim',
+  another: 'one',
+};
+
+function transformStateWithClones(currentState, actions) {
+  const history = [];
+  let clonedState = Object.assign({}, currentState);
+
+  actions.forEach((obj) => {
+    if (obj.type === 'clear') {
+      clonedState = {};
+      history.push({});
+    }
+
+    if (obj.type === 'addProperties') {
+      clonedState = { ...clonedState, ...obj.extraData };
+      history.push({ ...clonedState });
```

Comment:
Similar to the previous comment, avoid pushing the `clonedState` inside the 'addProperties' case. You should push the `clonedState` to the history array once at the end of the loop cycle. [DONT REPEAT YOURSELF]